dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/McpServlet.java

```diff
@@ -143,6 +155,9 @@ private void callTool(McpTransport transport) {
         } catch (IllegalArgumentException e) {
             throw new JSONRPCException(JSONRPCErrorCode.INVALID_PARAMS, List.of("Incorrect arguments in params"));
         } finally {
+            if (requestId != null && connection.isOngoingRequest(requestId)) {
+                connection.deregisterOngoingRequest(requestId);
+            }
```

Comment:
```suggestion
            connection.deregisterOngoingRequest(requestId);
```
`requestId` cannot be `null`

Attempting to de-register a request which is not ongoing has no effect, so we don't need to check it first.

Reply:
Thanks, done